pageserver/src/bin/pageserver.rs

```diff
@@ -115,7 +115,14 @@ fn main() -> Result<()> {
                     option_line
                 )
             })?;
+
             for (key, item) in doc.iter() {
+                if key == "id" {
+                    anyhow::ensure!(
```

Comment:
NIT: Should we rather WARN log it and skip it instead of failing?

Reply:
In my opinion node id sort of related to correctness so I would rather fail loudly. It is an error to have one id in config and a pageserver actually running with different id. If you need to forcibly change the value you can edit the config manually.